internal/rpc/starknet/types.go

```diff
@@ -490,3 +490,65 @@ func (r *RpcFelt) UnmarshalJSON(data []byte) error {
 func (r *RpcFelt) Felt() *felt.Felt {
 	return new(felt.Felt).SetHex(string(*r))
 }
+
+type StorageDiffItem struct {
+	Address string `json:"address"`
+	Key     string `json:"key"`
+	Value   string `json:"value"`
+}
+
+type DeployedContractItem struct {
+	Address   string `json:"address"`
+	ClassHash string `json:"class_hash"`
+}
+
+type DeclatedContractItem struct {
+	ClassHash string `json:"class_hash"`
+}
+
+type StateDiff struct {
+	StorageDiffs      []*StorageDiffItem      `json:"storage_diffs"`
+	DeployedContracts []*DeployedContractItem `json:"deployed_contracts"`
+	DeclaredContracts []*DeclatedContractItem `json:"declared_contracts"`
+}
+
+type StateUpdate struct {
+	BlockHash string     `json:"block_hash"`
+	NewRoot   string     `json:"new_root"`
+	OldRoot   string     `json:"old_root"`
+	StateDiff *StateDiff `json:"state_diff"`
+}
```

Comment:
Is it possible to move these types to the `types` package and delete the types that are currently there? Having them both appears redundant, but these seem more well-written.

Reply:
I agree and it's something that I noticed too, but that means making some changes to the sync process and isn't related to the RPC. I think will be better if we can do that on other PR later